sdk/go/common/workspace/plugins.go

```diff
@@ -344,6 +348,98 @@ func (source *gitlabSource) URL() string {
 	return fmt.Sprintf("gitlab://%s/%s", source.host, source.project)
 }
 
+type gitSource struct {
+	url  string
+	path string
+
+	cloneAndCheckout func(context.Context, string, plumbing.Hash, string) error
+	cloneOrPull      func(context.Context, string, plumbing.ReferenceName, string, bool) error
+}
+
+func newGitHTTPSSource(url *url.URL) (*gitSource, error) {
+	url.Scheme = "https"
+	u, path, err := gitutil.ParseGitRepoURL(url.String())
+	if err != nil {
+		return nil, err
+	}
+	if path != "" {
+		return nil,
+			// TODO[pulumi/pulumi#18250]: We currently don't support installing plugins from subdirectories.
+			//nolint:lll
+			errors.New("cannot install git plugin from subdirectory.  See https://github.com/pulumi/pulumi/issues/18250")
+	}
+	return &gitSource{
+		url:              u,
+		path:             path,
+		cloneAndCheckout: gitutil.GitCloneAndCheckoutCommit,
+		cloneOrPull:      gitutil.GitCloneOrPull,
+	}, nil
+}
+
+func (source *gitSource) GetLatestVersion(
+	ctx context.Context, _ func(*http.Request) (io.ReadCloser, int64, error),
+) (*semver.Version, error) {
+	// TODO: support this for the users convenience
+	return &semver.Version{},
```

Comment:
`nil`? Or does it have to be a non-nil pointer?

Reply:
Yeah it can be nil.  Changed it to that now.